app/gui2/src/stores/project.ts

```diff
@@ -22,7 +22,7 @@ export const useProjectStore = defineStore('project', () => {
   watchEffect((onCleanup) => {
     // For now, let's assume that the websocket server is running on the same host as the web server.
     // Eventually, we can make this configurable, or even runtime variable.
-    let socketUrl = location.origin.replace(/^http/, 'ws') + '/room'
+    const socketUrl = location.origin.replace(/^http/, 'ws') + '/room'
```

Comment:
kinda cursed ngl. i'd probably do `new URL(location.href)`, and then mutate `pathName` and `protocol` - not that it's necessarily better